File: problems/2215_find_the_difference_of_two_arrays/Solution.java
import java.util.*;

class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> s1 = new HashSet<>(), s2 = new HashSet<>();
        for (int n : nums1) s1.add(n);
        for (int n : nums2) s2.add(n);
        List<Integer> r1 = new ArrayList<>(), r2 = new ArrayList<>();
        for (int n : s1) if (!s2.contains(n)) r1.add(n);
        for (int n : s2) if (!s1.contains(n)) r2.add(n);
        return List.of(r1, r2);
    }
}

File: problems/2215_find_the_difference_of_two_arrays/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        set<int> s1(nums1.begin(), nums1.end()), s2(nums2.begin(), nums2.end());
        vector<int> r1, r2;
        for (int n : s1) if (!s2.count(n)) r1.push_back(n);
        for (int n : s2) if (!s1.count(n)) r2.push_back(n);
        return {r1, r2};
    }
};
